### ISSUE TITLE
[TimePicker] How to set minutes current date after select hour

### ISSUE THREAD

**Issue by Kwunnnnn** [mui/mui-x#10316]
TimePicker suggests minutes to '00' after selecting the hour row but '00' has disabled minutes. I want to not set default minutes at '00' I want the current minute. How can I do it? 

Example:  https://codesandbox.io/s/mui-x-issue-ls7npj

Expect: I want to select the hour as '19' and minutes auto-select at '30'

**Comment by flaviendelangle** [comment]
Hi,

You can use the `referenceDate` prop to change the default minutes (see [this example](https://codesandbox.io/s/mui-x-issue-forked-fhk7rn?file=/src/App.js))

But the default behavior should probably be smarter (cc @LukasTy)

**Comment by Kwunnnnn** [comment]
> 

Thank you, It's work 🚀🚀🚀

**Comment by flaviendelangle** [comment]
I'll keep the issue open to track the progress of the built-in support (if we want to improve it)

**Comment by Kwunnnnn** [comment]
I have one more question. If I want to reset row minute after select new hour value, How can I do?

**Comment by flaviendelangle** [comment]
You could probably analyse the value passed to `onChange` and when the hour changes between the old and the new value, you reset the minutes.
But there is no specific API to help you do that :+1:

**Comment by LukasTy** [comment]
> But the default behavior should probably be smarter (cc @LukasTy)

Good point. We could add a similar improvement as we have with the `DateCalendar`, where it will look for a closest non-disabled date when navigating the calendar.

**Comment by flaviendelangle** [comment]
That would be a great improvement :ok_hand:
